protocols/357404/357404.ot2.apiv2.py

```diff
@@ -0,0 +1,518 @@
+        reset, the protocol will start picking up tips from the first tip
+        box as defined in the slot order when assigning the labware definition
+        for that tip box. `pick_up()` will track tips for both pipettes if
+        applicable.
+
+        :param pipette: The pipette desired to pick up tip
+        as definited earlier in the protocol (e.g. p300, m20).
+        """
+        try:
+            pipette.pick_up_tip()
+        except protocol_api.labware.OutOfTipsError:
+            flash_lights()
+            ctx.pause("Replace empty tip racks")
+            pipette.reset_tipracks()
+            pipette.pick_up_tip()
+    # helper functions
+    '''
+    Define any custom helper functions outside of the pipette scope here, using
+    the convention seen above.
+
+    e.g.
+
+    def remove_supernatant(vol, index):
+        """
+        function description
+
+        :param vol:
+
+        :param index:
+        """
+
+
+    '''
+    class VolTracker:
+        def __init__(self, labware, well_vol,
+                     start=1, end=8,
+                     mode='reagent',
+                     pip_type='single',
+                     msg='Reset labware volumes'):
+            """
+            Voltracker tracks the volume(s) used in a piece of labware
+
+            :param labware: The labware to track
+            :param well_vol: The volume of the liquid in the wells, if using a
+            multi-pipette with a well plate, treat the plate like a reservoir,
+            i.e. start=1, end=1, well_vol = 8 * vol of each individual well.
+            :param pip_type: The pipette type used 'single' or 'multi'
+            :param mode: 'reagent' or 'waste'
+            :param start: The starting well
+            :param end: The ending well
+            :param msg: Message to send to the user when all wells are empty
+            (or full when in waste mode)
+
+            """
+            self.labware_wells = dict.fromkeys(
+                labware.wells()[start-1:end], 0)
+            self.labware_wells_backup = self.labware_wells.copy()
+            self.well_vol = well_vol
+            self.pip_type = pip_type
+            self.mode = mode
+            self.start = start
+            self.end = end
+            self.msg = msg
+
+            # Parameter error checking
+            if not (pip_type == 'single' or pip_type == 'multi'):
+                raise Exception('Pipette type must be single or multi')
+
+            if not (mode == 'reagent' or mode == 'waste'):
+                raise Exception('mode must be reagent or waste')
+
+        def flash_lights(self):
+            """
+            Flash the lights of the robot to grab the users attention
+            """
+            initial_light_state = ctx.rail_lights_on
+            opposite_state = not initial_light_state
+            for _ in range(5):
+                ctx.set_rail_lights(opposite_state)
+                time.sleep(0.5)
```

Comment:
ctx.delay(0.5) instead of time.sleep(0.5)

Reply:
Why didn't I think of that! thanks